test/functionalTest/cases/3458_legacy-forwarding-with-idpattern/legacy-forwarding-with-idpattern.test

```diff
@@ -238,7 +238,7 @@ Content-Length: 2
 
 04. Dump accumulator requests
 =============================
-POST http://localhost:REGEX(\d+)/v1/queryContext
+POST http://localhost:REGEX(\d+)/v1/queryContext?limit=20&offset=0&options=count
```

Comment:
Note that in /v1, count is not requested using `&options=count` but `&details=on`. Moreover, the result is not provided in `fiware-total-count` header but within the NGSIv1 responde payload, within `details` field. For instance:

```
    "errorCode": {
        "code": "200",
        "details": "Count: 4",
        "reasonPhrase": "OK"
    }
```

See for instance cases/0117_convop_using_standard_ops/getAllContextEntities.test

Fixing this properly will require additional modifications to the code. Taking into account that NGSIv1 is deprecated, I'd suggest to leave NGSIv1 CPrs (i.e. CPrs which implement `POST /v1/queryContext` endpoint instead of `POST /v2/query`) outside the scope of the CPr pagination functionality.

@ArqamFarooqui110719 what do you think? Are all the CPrs in your use case NGSIv2-based?

Reply:
> Note that in /v1, count is not requested using `&options=count` but `&details=on`. Moreover, the result is not provided in `fiware-total-count` header but within the NGSIv1 responde payload, within `details` field. For instance:
> 
Yes, I also have same understanding. Changes were done by mistake.
> 
> 
> @ArqamFarooqui110719 what do you think? Are all the CPrs in your use case NGSIv2-based?

We are discussing internally.

Reply:
> Note that in /v1, count is not requested using `&options=count` but `&details=on`. Moreover, the result is not provided in `fiware-total-count` header but within the NGSIv1 responde payload, within `details` field. For instance:
> 
> ```
>     "errorCode": {
>         "code": "200",
>         "details": "Count: 4",
>         "reasonPhrase": "OK"
>     }
> ```
> 
> See for instance cases/0117_convop_using_standard_ops/getAllContextEntities.test
> 
> Fixing this properly will require additional modifications to the code. Taking into account that NGSIv1 is deprecated, I'd suggest to leave NGSIv1 CPrs (i.e. CPrs which implement `POST /v1/queryContext` endpoint instead of `POST /v2/op/query`) outside the scope of the CPr pagination functionality.
> 
> @ArqamFarooqui110719 what do you think? Are all the CPrs in your use case NGSIv2-based?

Yes, I also have same understanding. We have discussed internally and decided that there is no need to fix for NGSIv1 forwarding.

Reply:
> Note that in /v1, count is not requested using `&options=count` but `&details=on`. Moreover, the result is not provided in `fiware-total-count` header but within the NGSIv1 responde payload, within `details` field. For instance:
> 
To resolve the v1 related failed test cases (for e.g. [0715_FT](https://github.com/telefonicaid/fiware-orion/pull/4275/commits/1e47b077ce86f15d09438331b764710751bf1910#diff-226f4665891635d2220e382009a930d80c82969be6986f38fb4a39513bc7104d)), I think there should be a check in [httpRequestSend.cpp](https://github.com/telefonicaid/fiware-orion/pull/4275/commits/1e47b077ce86f15d09438331b764710751bf1910#diff-d4292f348507898aae1bf4f36986d4e14a976e872a5d75c32a21403cebfb6651) at L#580 as below
FROM:
```
if ((providerLimit >= 0) && (providerOffset >=0))
{}
```
TO:
```
if ((apiVersion == V2) && (providerLimit >= 0) && (providerOffset >=0))
{}

```
In that case, there will not any need to change the v1 related FTs i.e.:

- test/functionalTest/cases/0715_forward_xauth_token_to_cprs/propagate_xauth_token_for_update_and_query_standard_ops.test
- test/functionalTest/cases/3458_legacy-forwarding-with-idpattern/legacy-forwarding-with-idpattern.test

In that case, we need to pass `ciP->apiVersion` [from postQueryContextL.cpp](https://github.com/telefonicaid/fiware-orion/pull/4275/commits/cf74832b410414a5abfaa662e1c053d29d00e2ae#diff-94987134f182c2d9887707e37d2ea64ec7bb98cbb40df87ddb5e5fc917778cd0) to `httpRequestSend.cpp`.
```
int httpRequestSend
(
   CURL*                                      _curl,
   const std::string&                         idStringForLogs,
   const std::string&                         from,
   ...
   std::string*                               outP,
   long long*                                 statusCodeP,
   const std::map<std::string, std::string>&  extraHeaders,
   int                                        providerLimit,
   int                                        providerOffset,
+  ApiVersion                                 apiVersion,
   const std::string&                         acceptFormat,
   long                                       timeoutInMilliseconds
)
```
@fgalan, What is your opinion on this?

Reply:
I'd try to avoid to include apiVersion as new paramter in httpRequestSend() as the function of this function should be independent of the api version to use. Maybe the api check should be earlier in the program logic...

I'll have a look more in detail.